tensorflow_io/core/BUILD

```diff
@@ -572,6 +572,7 @@ cc_binary(
         "@local_config_tf//:tf_header_lib",
     ] + select({
         "@bazel_tools//src/conditions:windows": [],
+        "@bazel_tools//src/conditions:darwin": [],
```

Comment:
The macOS failure is actually caused by this line. Let me think how to address it while skipping the libmemcached for macOS.